plasma_framework/contracts/src/transactions/eip712Libs/PaymentEip712Lib.sol

```diff
@@ -0,0 +1,120 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "../PaymentTransactionModel.sol";
+import "../outputs/PaymentOutputModel.sol";
+import "../../utils/UtxoPosLib.sol";
+
+library PaymentEip712Lib {
```

Comment:
@pnowosie do you mind to help me quickly do a sanity check on the implementation here?
I am porting the `Eip712StructHash`. Some changes:
1. add `txType` to Transaction
1. In Output, owner was `address` --> `bytes32` to be compatible with `outputGuard` field.

Reply:
:heavy_check_mark:  LGTM. 

Just one optimization hint: we can save on computing empty in/outputs hashes precomputing them in `Constants`. Also we already guarantee that in/outputs don't intersperse with empty one - so loops in _txHash could be optimized as well.

It is You to decide going for :point_up: or as further improvement.